From 50f0f79cefc993127f027f60f23e0cf6a16b122d Mon Sep 17 00:00:00 2001
From: Anthony Koo <Anthony.Koo@amd.com>
Date: Tue, 10 May 2016 17:54:18 -0400
Subject: [PATCH 0475/1722] drm/amd/dal: Fix soft hang when update pending bit
 not cleared

Fix issue where GRPH_SURFACE_UPDATE_PENDING bit
is not cleared because surface address is programmed while
CRTC is blanked.

Change sequence to unblank first.

Signed-off-by: Anthony Koo <Anthony.Koo@amd.com>
Acked-by: Jordan Lazare <Jordan.Lazare@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
index 47d203d..39f3966 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
@@ -1463,13 +1463,13 @@ static void update_plane_addrs(struct core_dc *dc, struct resource_context *res_
 					PIPE_LOCK_CONTROL_SURFACE,
 					false);
 
-		if (surface->public.flip_immediate)
-			pipe_ctx->mi->funcs->wait_for_no_surface_update_pending(pipe_ctx->mi);
-
 		if (!pipe_ctx->tg->funcs->set_blank(pipe_ctx->tg, false)) {
 			dm_error("DC: failed to unblank crtc!\n");
 			BREAK_TO_DEBUGGER();
 		}
+
+		if (surface->public.flip_immediate)
+			pipe_ctx->mi->funcs->wait_for_no_surface_update_pending(pipe_ctx->mi);
 	}
 }
 
-- 
2.7.4

